frontend/src/bundles/chat/pages/chat-modal.tsx

```diff
@@ -0,0 +1,63 @@
+import {
+    Modal,
+    ModalCloseButton,
+    ModalContent,
+    ModalOverlay,
+} from '~/bundles/common/components/components.js';
+import {
+    useAppDispatch,
+    useAppSelector,
+    useCallback,
+} from '~/bundles/common/hooks/hooks.js';
+
+import { Chat } from '../components/components.js';
+import { actions as chatActions } from '../store/chat.js';
+import { type GenerateTextRequestDto } from '../types/types.js';
+
+type Properties = {
+    isChatOpen: boolean;
+    onModalChatClose: () => void;
+};
+
+const ChatModal: React.FC<Properties> = ({ isChatOpen, onModalChatClose }) => {
```

Comment:
In this component, you should also send a request to delete if the user closes the tab. Here's how you can do it. But use chat-api instead of fetch. This is just an example with fetch.
`

          useEffect(() => {

            const handleBeforeUnload = async () => {
               
                await fetch('http://localhost:3001/api/v1/chat/ ', {
                    method: 'DELETE',
                    // You should modify http-api, because when you work with chat you should have:  credential:'include'.
                    // When you handle the 'beforeupload' event, you should add keepalive: true. So you should add it into http-api as well
                    credentials: 'include',
                    keepalive: true,
                }).catch((err) => {
                    console.log(err.message);
                });
            };
    
            window.addEventListener('beforeunload', handleBeforeUnload);

            return () => window.removeEventListener('beforeunload', handleBeforeUnload);
        }, []);
`

Reply:
@sergiy4 Thanks, implemented a hook to manage this